### PR TITLE
Volume PL +  Docker Engine Server integration

### PR DESCRIPTION
**Pull request by matthewavery**
This is the Handler code which connects the Volume portion of the storage layer to the docker engine server. The only thing missing is the extraconfig for populating the volumestore. This work will be included in a separate PR. The only major changes involve parameter validation and cache setup in the portlayer storage handlers.

### CODE REVIEW

**Comment by fdawg4l** [comment]
LGTM

File: lib/apiservers/engine/backends/volume.go [hunk 1]
```diff
@@ -28,9 +28,9 @@ import (
 	"github.com/vmware/vic/pkg/trace"
 )
 
-//NOTE: These might be moved to a utility package once there are multiple personalities
-const optsVolumeStoreKey string = "VolumeStore"
-const optsCapacityKey string = "Capacity"
+//NOTE: FIXME: These might be moved to a utility package once there are multiple personalities
+const OptsVolumeStoreKey string = "VolumeStore"
+const OptsCapacityKey string = "Capacity"
 const dockerMetadataModelKey string = "DockerMetaData"
```
Comment (caglar10ur) [inline lib/apiservers/engine/backends/volume.go]: use
```
const (
...
) 
```
format please

File: lib/apiservers/portlayer/restapi/handlers/storage_handlers.go [hunk 2]
```diff
@@ -237,3 +281,16 @@ func convertImage(image *spl.Image) *models.Image {
...
 	}
 }
+
+//utility functions
+
+func volumeToCreateResponse(volume *spl.Volume, model *models.VolumeRequest) models.VolumeResponse {
+	response := models.VolumeResponse{
+		Driver:   model.Driver,
+		Name:     volume.ID,
+		SelfLink: volume.SelfLink.String(), //FIXME: is this right?
```
Comment (caglar10ur) [inline lib/apiservers/portlayer/restapi/handlers/storage_handlers.go]: This is a very scary FIXME right there. Why do we have it, what's not clear and what can we do to clarify?

Comment (matthewavery) [inline lib/apiservers/portlayer/restapi/handlers/storage_handlers.go]: I am renaming that to label. and passing the label into it. There was some decision making to do on what to return from the portlayer.

File: lib/apiservers/portlayer/swagger.yml [hunk 3]
```diff
@@ -249,6 +249,8 @@ paths:
             $ref: "#/definitions/VolumeResponse"
         '500':
           description: "Error"
+          $schema:
+            $ref: "#/definitions/Error"
         default:
           description: "error"
```
Comment (sflxn) [inline lib/apiservers/portlayer/swagger.yml]: I know this wasn't added as part of this PR, but is a total waste.  It won't do what it was originally meant to do (capture generated client errors).  Defining a catch all error without an http status code feels very wrong to me.

Comment (fdawg4l) [inline lib/apiservers/portlayer/swagger.yml]: The status code is 500 @sflxn.  And as long as the response is relevant, I don't see the code as being a problem unless you want to map all possible error codes (http://www.gnu.org/software/libc/manual/html_node/Error-Codes.html) to an http error code.

In any case, we're not going to address this here.  Please create an issue and describe an alternative.

**Comment by gigawhitlocks** [comment]
LGTM

**Comment by matthewavery** [comment]
♻️

**Pull request merged**